src/main/java/uk/gov/hmcts/reform/finrem/caseorchestration/handler/UploadContestedCaseDocumentsAboutToSubmitHandler.java

```diff
@@ -74,6 +95,51 @@ public GenericAboutToStartOrSubmitCallbackResponse<Map<String, Object>> handle(C
         return response;
     }
 
+    private String getActiveUser(Long caseId, String userAuthorisation) {
+        String logMessage = "Logged in user role {} caseId {}";
+        String activeUserCaseRole = getActiveUserCaseRole(String.valueOf(caseId), userAuthorisation);
+        if (activeUserCaseRole.contains(CaseRole.APP_SOLICITOR.getValue())) {
+            log.info(logMessage, APPLICANT, caseId);
+            return APPLICANT;
+        } else if (activeUserCaseRole.contains(CaseRole.RESP_SOLICITOR.getValue())) {
+            log.info(logMessage, RESPONDENT, caseId);
+            return RESPONDENT;
+        } else if (activeUserCaseRole.contains(CaseRole.INTVR_SOLICITOR_1.getValue())) {
+            log.info(logMessage, INTERVENER_ONE, caseId);
+            return INTERVENER_ONE;
+        } else if (activeUserCaseRole.contains(CaseRole.INTVR_SOLICITOR_2.getValue())) {
+            log.info(logMessage, INTERVENER_TWO, caseId);
+            return INTERVENER_TWO;
+        } else if (activeUserCaseRole.contains(CaseRole.INTVR_SOLICITOR_3.getValue())) {
+            log.info(logMessage, INTERVENER_THREE, caseId);
+            return INTERVENER_THREE;
+        } else if (activeUserCaseRole.contains(CaseRole.INTVR_SOLICITOR_4.getValue())) {
+            log.info(logMessage, INTERVENER_FOUR, caseId);
+            return INTERVENER_FOUR;
+        }
+        return activeUserCaseRole;
+    }
+
+
+    public String getActiveUserCaseRole(final String caseId, final String userAuthorisation) {
```

Comment:
Can we please put this method into idamService as its related to idam to get user role

Reply:
done, this is not exactly idam service method, moved to AssignCaseAccessService.